Automattic/jetpack
# Connection Manager UI

#### Changes proposed in this Pull Request:
The Connection UI

#### Jetpack product discussion
p9dueE-1vM-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Using plugins like Client Example and Jetpack Boost to test it will require symlinking the latest version of `jetpack-connection` from those plugins.
An alternative would be to create a plugin with following code, which will emulate having two plugins that use the connection:

<details>
<summary>Unfold the plugin</summary>

```php
<?php
/** Plugin Name: Jetpack Connection Manager Test 1 */

use Automattic\Jetpack\Config;

add_action( 'plugins_loaded', function() {
	$config = new Config();
	$config->ensure(
		'connection',
		array(
			'slug'     => 'plugin-1',
			'name'     => 'Plugin 1',
			'url_info' => 'https://github.com/Automattic/jurassic-tube-backend'
		)
	);

	$config = new Config();
	$config->ensure(
		'connection',
		array(
			'slug'     => 'plugin-2',
			'name'     => 'Plugin 2',
			'url_info' => 'https://github.com/Automattic/jurassic-tube'
		)
	);
}, 1 );
```
</details>

**This is the UI you should see:**
<img width="420" alt="Screen Shot 2021-01-08 at 4 55 24 PM" src="https://user-images.githubusercontent.com/1341249/104200786-aa55be80-53ee-11eb-87a4-d35f2973908e.png">


1. Connect Jetpack, go to "Tools -> Connection Manager".
2. Click "Refresh Connection". Confirm that the button has changed the label and no longer clickable. You will be taken to the Calypso flow for reconnecting (no in-place flow yet).
3. After reconnecting you'll be redirected back to Connection Manager. The "Refresh Connection" button should become green for 5 seconds.
4. Use the "Broken Token" plugin to set an invalid blog token.
5. Click "Refresh Connection" again. The button should be disabled, and then turn green. Use "Broken Token" plugin to confirm that the blog token has been restored.
6. Click "Disconnect" for a plugin. The button label should change to "Connect".
7. Reload the page, confirm that the disconnected state persists for that plugin.
8. If you're testing on actual plugins (e.g. Client Example or Jetpack Boost), try "disconnecting" them from Connection Manager and confirm that they get softly disconnected.
9. Try connecting them back, and confirm that the plugins are connected again with no need to go through the connection flow.

#### Proposed changelog entry for your changes:
* User Interface for managing connection to the WordPress.com
